### PR TITLE
Updated dynamic_quantization_bert_tutorial.rst

### LINKED ISSUE #1780: [tutorials] 'ids_tensor' function is not exist in tutorial of 'DYNAMIC QUANTIZATION ON BERT'
In this [tutorials](https://pytorch.org/tutorials/intermediate/dynamic_quantization_bert_tutorial.html#serialize-the-quantized-model), `ids_tesnor` function is not exist. So following code needs to be added.
```python
import random 
global_rng = random.Random()

def ids_tensor(shape, vocab_size, rng=None, name=None):
    #  Creates a random int32 tensor of the shape within the vocab size
    if rng is None:
        rng = global_rng

    total_dims = 1
    for dim in shape:
        total_dims *= dim

    values = []
    for _ in range(total_dims):
        values.append(rng.randint(0, vocab_size - 1))

    return torch.tensor(data=values, dtype=torch.long, device='cpu').view(shape).contiguous()
```
The code referred to this [link](https://pytorch.org/tutorials/prototype/graph_mode_dynamic_bert_tutorial.html#quantizing-bert-model).

cc @jerryzh168 @z-a-f @vkuzo

### PR DESCRIPTION
**Pull request by Srujana2505**
Fixes #1780 

## Description

- Added ids_tensor function to the doc
- Corrected the commands for the new versions of scikit-learn and transformers(hugging face) required to test with latest pytorch version
- while creating a traced model, I am finding an error that the format needs to be changed to other data structure like list so added **strict = False**(In line 552)

## Checklist

- [x] The issue that is being fixed is referred in the description (see above "Fixes #ISSUE_NUMBER")
- [x] Only one issue is addressed in this pull request
- [x] Labels from the issue that this PR  #is fixing are added to this pull request
- [x] No unnessessary issues are included into this pull request.

cc @jerryzh168 @z-a-f @vkuzo

### CODE REVIEW

File: intermediate_source/dynamic_quantization_bert_tutorial.rst [hunk 1]
```diff
@@ -523,13 +523,33 @@ processing the evaluation of MRPC dataset.
 We can serialize and save the quantized model for the future use using
 `torch.jit.save` after tracing the model.
 
+.. code:: python
+
+    import random 
+    global_rng = random.Random()
+
+    def ids_tensor(shape, vocab_size, rng=None, name=None):
```
Comment (carljparker) [inline intermediate_source/dynamic_quantization_bert_tutorial.rst:531]: I don't see the `name=None` parameter being used anywhere . . . Is that necessary or can be delete for brevity?

**Review by carljparker — CHANGES_REQUESTED** [review]

File: intermediate_source/dynamic_quantization_bert_tutorial.rst (hunk 1 above)
Comment (Srujana2505) [reply intermediate_source/dynamic_quantization_bert_tutorial.rst:531]: Yeah that is true, let me check that once!

Comment (Srujana2505) [reply intermediate_source/dynamic_quantization_bert_tutorial.rst]: Can you please check the changes once! Also there was an other issue #1406 for the same tutorial in which the colab file needs changes. Can you point me out on how to get the colab edit access or do I need to make a new colab file from scratch?

File: intermediate_source/dynamic_quantization_bert_tutorial.rst [hunk 2]
```diff
@@ -528,8 +525,11 @@ We can serialize and save the quantized model for the future use using
 
 .. code:: python
 
-    def ids_tensor(shape, vocab_size, rng=None, name=None):
-        #  Creates a random int32 tensor of the shape within the vocab size
+    import random 
+    global_rng = random.Random()
+
+    def ids_tensor(shape, vocab_size, rng=None,):
```
Comment (NicolasHug) [inline intermediate_source/dynamic_quantization_bert_tutorial.rst:531]:
```suggestion
    def ids_tensor(shape, vocab_size, rng=None):
```

File: intermediate_source/dynamic_quantization_bert_tutorial.rst [hunk 3]
```diff
@@ -393,9 +393,6 @@ We reuse the tokenize and evaluation function from `Huggingface <https://github.
                                                     label_list=label_list,
                                                     max_length=args.max_seq_length,
                                                     output_mode=output_mode,
-                                                    pad_on_left=bool(args.model_type in ['xlnet']),                 # pad on the left for xlnet
-                                                    pad_token=tokenizer.convert_tokens_to_ids([tokenizer.pad_token])[0],
-                                                    pad_token_segment_id=4 if args.model_type in ['xlnet'] else 0,
```
[the remark this answers is not in the document]
Comment (Srujana2505) [reply intermediate_source/dynamic_quantization_bert_tutorial.rst:398]: With those arguments I am finding an error that says the glue_convert_examples_to_features doesn't take those arguments. So I removed them in reference with the above funtion implemented in hugging face. The link can be found [here.](https://github.com/huggingface/transformers/blob/main/src/transformers/data/processors/glue.py)

File: intermediate_source/dynamic_quantization_bert_tutorial.rst [hunk 4]
```diff
@@ -543,11 +543,14 @@ We can serialize and save the quantized model for the future use using
 
         return torch.tensor(data=values, dtype=torch.long, device='cpu').view(shape).contiguous()
 
+
+.. code:: python
+
```
[the remark this answers is not in the document]
Comment (Srujana2505) [reply intermediate_source/dynamic_quantization_bert_tutorial.rst:548]: Sure I'll update the above changes

File: intermediate_source/dynamic_quantization_bert_tutorial.rst (hunk 2 above)
Comment (Srujana2505) [reply intermediate_source/dynamic_quantization_bert_tutorial.rst:531]: With the new commit in main branch, the new ids_tensor function is :
**def ids_tensor(shape, vocab_size):
    #  Creates a random int32 tensor of the shape within the vocab size
    return torch.randint(0, vocab_size, shape=shape, dtype=torch.int, device='cpu')**
But I am getting an error which says: _randint() received an invalid combination of arguments - got (int, int, device=str, dtype=torch.dtype, shape=list)_
So do I proceed with the above function or should I use the one I'm currently modifying?

**Comment by svekars** [comment]
please resolve the conflict

File: intermediate_source/dynamic_quantization_bert_tutorial.rst [hunk 5]
```diff
@@ -532,7 +529,7 @@ We can serialize and save the quantized model for the future use using
     token_type_ids = ids_tensor([8, 128], 2)
     attention_mask = ids_tensor([8, 128], vocab_size=2)
     dummy_input = (input_ids, attention_mask, token_type_ids)
-    traced_model = torch.jit.trace(quantized_model, dummy_input)
+    traced_model = torch.jit.trace(quantized_model, dummy_input, strict=False)
```
[the remark this answers is not in the document]
Comment (Srujana2505) [reply intermediate_source/dynamic_quantization_bert_tutorial.rst:532]: Yeah I'll try that

[the remark this answers is not in the document]
Comment (Srujana2505) [reply intermediate_source/dynamic_quantization_bert_tutorial.rst:532]: image
I think the error occurs when we serialize and not when we save. So we still need to fix the serializing function. And in the error it is suggested to use strict=False. I am not sure of other ways to solve this.

[the remark this answers is not in the document]
Comment (Srujana2505) [reply intermediate_source/dynamic_quantization_bert_tutorial.rst:532]: Then do I need to remove the ids_tensor function and just use the previous function from the colab version?

**Comment by svekars** [comment]
Closing as stale.
